src/NetSapiensSharp/Connector.cs

```diff
@@ -67,6 +73,12 @@ private void Authenticate()
             }
         }
 
+        private bool IsAccessTokenExpired()
+        {
+            var myElapsedSeconds = Math.Ceiling((DateTime.UtcNow - _ExpirationStartTime).TotalSeconds);
+            return myElapsedSeconds >= (double)_ExpiresInSeconds;
```

Comment:
Talked to Dennis about this. We may want to add a 3 second window in case of network failure.
```suggestion
            return myElapsedSeconds + 3.00 >= (double)_ExpiresInSeconds;
```